tests/generic_config_updater/test_ntp.py

```diff
@@ -0,0 +1,204 @@
+import logging
+import pytest
+import re
+
+from tests.common.helpers.assertions import pytest_assert
+from tests.generic_config_updater.gu_utils import apply_patch, expect_op_failure, expect_op_success
+from tests.generic_config_updater.gu_utils import generate_tmpfile, delete_tmpfile
+from tests.generic_config_updater.gu_utils import create_checkpoint, delete_checkpoint, rollback_or_reload
+
+logger = logging.getLogger(__name__)
+
+pytestmark = [
+    pytest.mark.disable_loganalyzer,
+    pytest.mark.topology('any'),
+    pytest.mark.device_type('vs')
+]
+
+NTP_CONF         = "/etc/ntp.conf"
+NTP_SERVER_INIT  = "10.0.0.1"
+NTP_SERVER_DUMMY = "10.0.0.2"
+NTP_SERVER_RE    = "server {} iburst"
+
+@pytest.fixture(autouse=True)
+def setup_env(duthosts, rand_one_dut_hostname):
+    """
+    Setup/teardown fixture for ntp server config
+    Args:
+        duthosts: list of DUTs.
+        rand_selected_dut: The fixture returns a randomly selected DuT.
+    """
+    duthost = duthosts[rand_one_dut_hostname]
+    create_checkpoint(duthost)
+
+    init_ntp_servers = running_ntp_servers(duthost)
+
+    yield
+
+    try:
+        logger.info("Rolled back to original checkpoint")
+        rollback_or_reload(duthost)
+
+        cur_ntp_servers = running_ntp_servers(duthost)
+        pytest_assert(cur_ntp_servers == init_ntp_servers,
```

Comment:
>pytest_assert [](http://example.com/codeflow?start=8&length=13)

Did you check `ntp-config` service is just restarted?

Reply:
Config file `/etc/ntp.conf` will only change when ntp-config service is restarted. 
So I just check if the file being changed.